dlrover/python/master/diagnosis/diagnosis_manager.py

```diff
@@ -39,21 +49,92 @@
 class DiagnosisManager:
     """
     DiagnosisManager is to manage all diagnosis issues in a training job
+
     """
 
-    def __init__(self):
+    def __init__(self, job_name=None):
         self._is_observing_started = False
         self._data_manager: DiagnosisDataManager = DiagnosisDataManager(600)
         self._diagnostician: Diagnostician = Diagnostician(self._data_manager)
         self._job_context = get_job_context()
+        self._job_name = job_name
+        self._metric_monitor = None
+        self._xpu_type = ""
+        self._lock = threading.Lock()
 
     def collect_diagnosis_data(self, data: DiagnosisData):
         self._data_manager.store_data(data)
 
+    def collect_xpu_info(self, xpu_type):
+        with self._lock:
+            if not xpu_type:
+                return
+            if not self._xpu_type:
+                logger.info(f"the XPU accelerator is {xpu_type}")
+                self._xpu_type = xpu_type
+                self.start_metric_collect()
+
+    def get_xpu_info(self):
+        with self._lock:
+            return self._xpu_type
+
     def pre_check(self):
         logger.info("Start Diagnosis Manager to pre-check training...")
         pass
 
+    def start_metric_collect(self):
+        """
+        create a XpuMetricMonitor instance based on worker XPU type
+        start a thread to collect metrics data
+        store the data into global JobMetricContext
+
+        """
+        logger.info(f"start {self._xpu_type} metric collector...")
+        url = os.getenv("DLROVER_METRIC_URL", "")
+        if url == "":
+            logger.warning("no GPU metrics url defined, stop metric collector")
```

Comment:
The implementation of metric collection should be more general. The collected data should not be limited to XPU data, it should also include other general metrics. Just like the variable naming: DLROVER_METRIC_URL(not DLROVER_XPU_METRIC_URL).